openmoc/materialize.py

```diff
@@ -2,170 +2,356 @@
+
+    if geometry and 'openmoc.Geometry' not in str(type(geometry)):
+        py_printf('ERROR', 'Unable to load HDF5 file "%s" for "%s" which is not '
+                           'an OpenMOC Geometry', filename, str(type(geometry)))
+
+    # Instantiate dictionary to hold Materials to return to user
+    materials = {}
+    num_groups = int(f.attrs['# groups'])
+
+    # If a Geometry was passed in, extract all cells or materials from it
+    if geometry:
+        if domain_type == 'material':
+            domains = geometry.getAllMaterials()
+        elif domain_type == 'cell':
+            domains = geometry.getAllMaterialCells()
+        else:
+            py_printf('ERROR', 'Domain type "%s" is not supported', domain_type)
+
+    # Iterate over all domains (e.g., materials or cells) in the HDF5 file
+    for domain_spec in f[domain_type]:
+
+        py_printf('INFO', 'Importing cross sections for %s "%s"',
+                          domain_type, str(domain_spec))
+
+        # Create shortcut to HDF5 group for this domain
+        domain_group = f[domain_type][domain_spec]
+
+        # If domain_spec is an integer, it is an ID; otherwise a string name
+        if domain_spec.isdigit():
+            domain_spec = int(domain_spec)
+        else:
+            domain_spec = str(domain_spec)
+
+        # If using an OpenMOC Geometry, extract a Material from it
+        if geometry:
+
+            if domain_type == 'material':
+                material = _get_domain(domains, domain_spec)
+
+            elif domain_type == 'cell':
+                cell = _get_domain(domains, domain_spec)
+                material = cell.getFillMaterial()
+
+                # If the user filled the Cell with a Material, clone it
+                if material != None:
+                    material = material.clone()
+
+                # If the Cell does not contain a Material, creat one for it
+                else:
+                    if isinstance(domain_spec, int):
+                        material = openmoc.Material(id=domain_spec)
+                    else:
+                        material = openmoc.Material(name=domain_spec)
+
+                # Fill the Cell with the new Material
+                cell.setFill(material)
+
+        # If not Geometry, instantiate a new Material with the ID/name
+        else:
+            if isinstance(domain_spec, int):
+                material = openmoc.Material(id=domain_spec)
+            else:
+                material = openmoc.Material(name=domain_spec)
+
+        # Add material to the collection
+        materials[domain_spec] = material
+        material.setNumEnergyGroups(num_groups)
+
+        # Search for the total/transport cross section
+        if 'transport' in domain_group:
+            sigma = _get_numpy_array(domain_group, 'transport', suffix)
+            material.setSigmaT(sigma)
+            py_printf('INFO', 'Loaded "transport" MGXS for "%s %s"', 
```

Comment:
I believe this will error out if `domain_spec` is an int, so you should probably cast `domain_spec` as a str.